src/AI/WallyAI.java

```diff
@@ -0,0 +1,788 @@
+  {
+    myCo = co;
+    for( UnitModel um : myCo.unitModels )
+    {
+      for( WeaponModel wm : um.weaponModels )
+      {
+        if( wm.canFireAfterMoving )
+          mobileAttackRange = Math.max(mobileAttackRange, wm.maxRange);
+      }
+    }
+  }
+
+  @Override
+  public void initTurn(GameMap gameMap)
+  {
+    turnNum++;
+    log(String.format("[======== Wally initializing turn %s for %s =========]", turnNum, myCo));
+
+    // Make sure we don't have any hang-ons from last time.
+    actions.clear();
+
+    // Create a list of every property we don't own, but want to.
+    unownedProperties = new ArrayList<XYCoord>();
+    for( int x = 0; x < gameMap.mapWidth; ++x )
+    {
+      for( int y = 0; y < gameMap.mapHeight; ++y )
+      {
+        XYCoord loc = new XYCoord(x, y);
+        if( gameMap.getLocation(loc).isCaptureable() && myCo.isEnemy(gameMap.getLocation(loc).getOwner()) )
+        {
+          unownedProperties.add(loc);
+        }
+      }
+    }
+    capturingProperties = new ArrayList<XYCoord>();
+    for( Unit unit : myCo.units )
+    {
+      if( unit.getCaptureProgress() > 0 )
+      {
+        capturingProperties.add(unit.getCaptureTargetCoords());
+      }
+    }
+
+    // Check for a turn-kickoff power
+    AIUtils.queueCromulentAbility(actions, myCo, CommanderAbility.PHASE_TURN_START);
+  }
+
+  @Override
+  public void endTurn()
+  {
+    log(String.format("[======== Wally ending turn %s for %s =========]", turnNum, myCo));
+    logger = new StringBuffer();
+  }
+
+  private void log(String message)
+  {
+    System.out.println(message);
+    logger.append(message).append('\n');
+  }
+
+  @Override
+  /**
+   * Our to-do list for a given turn:
+   * 1) siege attacks
+   * 2) try for any mobile kills
+   * 3) capture if needed
+   * 4) travel
+   * 5) build
+   */
+  public GameAction getNextAction(GameMap gameMap)
+  {
+    GameAction nextAction = null;
+    do
+    {
+      // If we have more actions ready, don't bother calculating stuff.
+      if( !actions.isEmpty() )
+      {
+        GameAction action = actions.poll();
+        log("  Action: " + action);
+        return action;
+      }
+      else if( unitQueue.isEmpty() )
+      {
+        stateChange = false; // There's been no gamestate change since we last iterated through all the units, since we're about to do just that
+        for( Unit unit : myCo.units )
+        {
+          if( unit.isTurnOver )
+            continue; // No actions for stale units.
+          unitQueue.offer(unit);
+        }
+      }
+
+      Queue<Unit> tempQueue = new ArrayDeque<Unit>();
+      // Evaluate siege attacks
+      while (actions.isEmpty() && !unitQueue.isEmpty())
+      {
+        Unit unit = unitQueue.poll();
+        if( (gameMap.getEnvironment(unit.x, unit.y).terrainType == TerrainType.FACTORY) || !unit.model.hasImmobileWeapon() )
+        {
+          tempQueue.offer(unit);
+          continue;
+        }
+
+        // Find the possible destinations.
```

Comment:
Is this just very tongue-in-cheek, or was this comment copied from somewhere?

Reply:
Uhhh. Let's go with yes?

Reply:
"Possible Destinations? Right where I'm currently sitting. Done."

Reply:
I guess it'd be fair to remove the "s" then, eh?

Reply:
Hah, I guess.